test/unit/docsify.test.js

```diff
@@ -0,0 +1,133 @@
+/* global before after */
+/* eslint-disable no-global-assign */
+require = require('esm')(module /* , options */);
+
+const http = require('http');
+const handler = require('serve-handler');
+const { expect } = require('chai');
+const { initJSDOM } = require('../_helper');
+
+const docsifySite = 'http://127.0.0.1:3000';
+
+const markup = /* html */ `<!DOCTYPE html>
+  <html>
+    <head></head>
+    <body>
+      <div id="app"></div>
+      <script src="/lib/docsify.js"></script>
+    </body>
+  </html>
+`;
+
+/** @type {ReturnType<typeof http.createServer>} */
+let server;
+
+describe('Docsify public API', () => {
+  before(async () => {
+    server = http.createServer((request, response) => {
+      // You pass two more arguments for config and middleware
+      // More details here: https://github.com/zeit/serve-handler#options
+      return handler(request, response);
+    });
+
+    await new Promise(r => server.listen(3000, r));
+  });
+
+  after(async () => {
+    server.close(err => {
+      if (err) {
+        console.error(err); // eslint-disable-line
+        process.exit(1);
+      }
+      // eslint-disable-next-line
+      console.log('Server closed.');
+    });
+  });
+
+  it('global APIs are available', async () => {
+    // const DOM = new (require('jsdom').JSDOM)(markup, {
+    const DOM = initJSDOM(markup, {
+      url: docsifySite,
+      runScripts: 'dangerously',
+      resources: 'usable',
+    });
```

Comment:
Here we instantiate `JSDOM` with the `url` of `http://127.0.0.1:3000`. When it evaluates the HTML inside of `markup`, it will load the `lib/docsify.js` file from `http://127.0.0.1:3000/lib/docsify.js`.